### PR TITLE
Allow duplicate station name when country hint is enable and country different

### PR DESCRIPTION
**Pull request by noplay**
Instead of adding more & more station to the HOMONYM constant.

### CODE REVIEW

File: test_data.rb [hunk 1]
```diff
@@ -199,51 +191,13 @@ def test_unique_suggestable_name
...
-      homonym_station = STATIONS_BY_ID[homonym_id]
-      folded_homonym_name = fold(homonym_station["name"] || "")
-      has_homonym = stations.any? do |station|
-        station["id"] != homonym_station["id"] &&
-          station["folded_name"] == folded_homonym_name &&
-          station["suggestable?"]
-        end
-        assert has_homonym,
-          "Station #{homonym_station["name"]} (#{homonym_station["id"]}) does not have a suggestable homonym station"
+      assert !names.include?("#{row['name']} (#{row['country']})"), "Station #{name} (#{row["id"]}) has a name already used in this country"
```
Comment (avandecreme) [inline test_data.rb]: `refute` instead of `assert !`

File: test_data.rb [hunk 2]
```diff
@@ -199,51 +191,13 @@ def test_unique_suggestable_name
     names = Set.new
 
     SUGGESTABLE_STATIONS.each do |row|
-      if !Constants::HOMONYM_STATIONS.include?(row["id"])
-        assert !names.include?(row["name"]), "Station #{row["name"]} (#{row["id"]}) has a name already used"
-
+      name = row["name"]
+      if row["country_hint"] == "f"
+        assert !names.include?(row["name"]), "Station #{name} (#{row["id"]}) has a name already used"
```
Comment (avandecreme) [inline test_data.rb]: `refute` instead of `assert !`

File: test_data.rb [hunk 3]
```diff
@@ -199,51 +191,13 @@ def test_unique_suggestable_name
     names = Set.new
```
Comment (avandecreme) [inline test_data.rb]: Shouldn't you have two sets now? One to test `name` and one to test `name + country`?

Comment (noplay) [reply test_data.rb]: Both work, but I had amend the commit

**Comment by noplay** [comment]
Commit amend

File: test_data.rb (hunk 2 above)
Comment (noplay) [reply test_data.rb]: Fixed

File: test_data.rb (hunk 1 above)
Comment (noplay) [reply test_data.rb]: Fixed

**Pull request merged**